projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1856,8 +1856,8 @@ export abstract class IgxGridBaseDirective implements GridType,
             }
 
             value.type = FilteringExpressionsTreeType.Regular;
-            if (value && this.columns) {
-                this._filteringExpressionsTree = recreateTreeFromFields(value, this.columns) as IFilteringExpressionsTree;
+            if (value && this._columns && this._columns.length > 0) {
```

Comment:
Also could be just `this._columns?.length` but by the looks of it it's always initialized and not checked elsewhere, so:
```suggestion
            if (value && this._columns.length > 0) {
```